### PR TITLE
LPS-165547 Fixed Scroll overlay, outside Click of breadcrumb, inside input key behavior,set breadcrumb item context

### PR DESCRIPTION
**Pull request by welitonpe**

### CODE REVIEW

**Comment by welitonpe** [comment]
ci:test:sf

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/components/BreadcrumbFinder/index.tsx [hunk 1]
```diff
@@ -41,14 +78,17 @@ const BreadcrumbFinder = () => {
 			className="breadcrumb-finder-container breadcrumb-finder-navigator"
 			id="breadcrumbFinderContainer"
 		>
-			<div className="breadcrumb-finder-overlay"></div>
+			<div
+				className="breadcrumb-finder-overlay overlay"
+				onClick={() => setActive((previus) => !previus)}
+			></div>
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/components/BreadcrumbFinder/index.tsx:84]:
```suggestion
			<div
				className="breadcrumb-finder-overlay overlay"
				onClick={() => setActive((active) => !active)}
			/>
```

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/components/BreadcrumbFinder/index.tsx [hunk 2]
```diff
@@ -41,14 +78,17 @@ const BreadcrumbFinder = () => {
 			className="breadcrumb-finder-container breadcrumb-finder-navigator"
 			id="breadcrumbFinderContainer"
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/components/BreadcrumbFinder/index.tsx:79]: Maybe remove this id

```suggestion
			className="breadcrumb-finder-container breadcrumb-finder-navigator"
			id="breadcrumbFinderContainer"
```

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/components/BreadcrumbFinder/index.tsx [hunk 3]
```diff
@@ -81,6 +121,7 @@ const BreadcrumbFinder = () => {
 									if (key === 'Backspace' && search === '') {
 										onBackscape();
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/components/BreadcrumbFinder/index.tsx:122]: Early return

```suggestion
										return onBackscape();
```

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/components/BreadcrumbFinder/index.tsx [hunk 4]
```diff
@@ -90,17 +131,19 @@ const BreadcrumbFinder = () => {
 						</div>
 					</div>
 
-					<hr></hr>
+					<hr />
 
-					<ul className="list-unstyled">
+					<ul className="list-unstyled" ref={ulRef}>
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/components/BreadcrumbFinder/index.tsx:136]:
```suggestion
					<ul className="list-unstyled" ref={listRef}>
```

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/components/BreadcrumbFinder/index.tsx [hunk 5]
```diff
@@ -90,17 +131,19 @@ const BreadcrumbFinder = () => {
...
 						{items.map((item, itemIndex) => (
 							<li
 								className={classNames(
 									'breadcrumb-finder-item cursor-pointer',
 									{active: itemIndex === index}
 								)}
+								id={item.label}
 								key={itemIndex}
 								onClick={() => onClickRow(itemIndex)}
+								ref={tabRefs[itemIndex]}
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/components/BreadcrumbFinder/index.tsx:146]:
```suggestion
								ref={listItemsRef[itemIndex]}
```

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/components/BreadcrumbFinder/useBreadcrumbFinder.ts [hunk 6]
```diff
@@ -12,18 +12,31 @@
...
+	useRef,
+	useState,
+} from 'react';
 import {useHotkeys} from 'react-hotkeys-hook';
 import {useNavigate} from 'react-router-dom';
 
 import useBreadcrumb, {defaultEntities} from '../../hooks/useBreadcrumb';
 
-const ON_CLICK_ROW_ENABLED = false;
+const ON_CLICK_ROW_ENABLED = true;
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/components/BreadcrumbFinder/useBreadcrumbFinder.ts:28]: This line can be removed

```suggestion

```

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/components/BreadcrumbFinder/useBreadcrumbFinder.ts [hunk 7]
```diff
@@ -35,31 +48,66 @@ const useBreadcrumbFinder = () => {
...
 
+	const ulRef = useRef<HTMLUListElement>(null);
+
+	const tabRefs = useMemo(
+		() =>
+			Array.from({length: items.length}).map(() =>
+				createRef<HTMLLIElement>()
+			),
+		[items.length]
+	);
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/components/BreadcrumbFinder/useBreadcrumbFinder.ts:59]:
```suggestion
	const listItemsRef = useMemo(
		() =>
			Array.from({length: items.length}).map(() =>
				createRef<HTMLLIElement>()
			),
		[items.length]
	);
```

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/components/BreadcrumbFinder/useBreadcrumbFinder.ts [hunk 8]
```diff
@@ -35,31 +48,66 @@ const useBreadcrumbFinder = () => {
...
 			setActive(false);
 		}
 	};
 
+	const scrollIntoView = useCallback((liElement: any) => {
+		if (liElement.current) {
+			ulRef.current?.scrollTo({
+				behavior: 'smooth',
+				left: 0,
+				top: liElement.current.offsetTop - 90,
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/components/BreadcrumbFinder/useBreadcrumbFinder.ts:106]: Create variable for this magic number, 90

```suggestion
				top: liElement.current.offsetTop - 90,
```

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/components/BreadcrumbFinder/useBreadcrumbFinder.ts [hunk 9]
```diff
@@ -80,31 +130,68 @@ const useBreadcrumbFinder = () => {
...
+			scrollIntoView(tabRefs[currentIndex]);
+
 			return currentIndex;
 		});
-	}, [itemsLength]);
+	}, [itemsLength, scrollIntoView, tabRefs]);
+
+	const onEscape = useCallback(() => {
+		setActive(false);
+	}, []);
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/components/BreadcrumbFinder/useBreadcrumbFinder.ts:141]: Simplify

```suggestion
	const onEscape = useCallback(() => setActive(false), []);
```

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/components/BreadcrumbFinder/useBreadcrumbFinder.ts [hunk 10]
```diff
@@ -80,31 +130,68 @@ const useBreadcrumbFinder = () => {
...
 
 	const onTab = useCallback(() => {
 		setBreadCrumb((prevBreadCrumb) => [...prevBreadCrumb, activeItem]);
 		setIndex(0);
 		setSearch('');
-		inputRef.current?.focus();
+		setTimeout(() => {
+			inputRef.current?.focus();
+		}, 10);
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/components/BreadcrumbFinder/useBreadcrumbFinder.ts:149]:
```suggestion
		}, CREATE_VARIABLE_FOR_THIS);
```

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/hooks/useBreadcrumb.ts [hunk 11]
```diff
@@ -150,11 +149,13 @@ const defaultEntities: Entity[] = [
...
 	{
 		entity: 'caseresults',
+		getPage: ([projectId, routineId, buildId, caseResultsId]) =>
+			`/project/${projectId}/routines/${routineId}/build/${buildId}/case-result/${caseResultsId}`,
 		getResource: ([, , buildId]) =>
 			`/caseresults?filter=${searchUtil.eq(
 				'buildId',
 				buildId
-			)}&nestedFields=case&pageSize=1000`,
+			)}&nestedFields=case,r_runToCaseResult_c_runId&pageSize=1000`,
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/hooks/useBreadcrumb.ts:158]:
```suggestion
			)}&nestedFields=case&pageSize=1000`,
```

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/styles/components/_breadcrumb_finder.scss [hunk 12]
```diff
@@ -106,13 +106,32 @@
...
+	position: fixed;
+}
+
+[aria-hidden='true'] {
+	display: none;
+}
+[aria-hidden='false'] {
+	display: block;
+}
+
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/styles/components/_breadcrumb_finder.scss:121]: I think this is not scoped by any css

**Review by kevenleone — CHANGES_REQUESTED** [review]
LGTM 🔥

**Pull request closed without merging**